src/Client.php

```diff
@@ -68,4 +68,11 @@ public function getTypes()
     {
         return $this->encoder->__getTypes();
     }
+
+    public function overrideLocation($location)
+    {
+        $copy = clone $this;
+        $copy->encoder = $this->encoder->overrideLocation($location);
+        return $copy;
+    }
```

Comment:
I like the concept of making this an immutable method :+1: How about naming this `withLocation()` to make this more obvious?

Also, can you add some documentation / add this to the README? :+1: